### ISSUE TITLE
Broken in 10.1.7

### ISSUE THREAD

**Issue by mxpxh2ogo** [ascott18/ImprovedStableFrame#5]
Frames are broken and addon is not working. This is error (no other addons on). Love the addon! Thanks!

Message: ...e/AddOns/ImprovedStableFrame/ImprovedStableFrame.lua:102: attempt to index global 'PetStableModel' (a nil value)
Time: Wed Sep  6 07:43:58 2023
Count: 1
Stack: ...e/AddOns/ImprovedStableFrame/ImprovedStableFrame.lua:102: attempt to index global 'PetStableModel' (a nil value)
[string "@Interface/AddOns/ImprovedStableFrame/ImprovedStableFrame.lua"]:102: in main chunk

Locals: clientVersion = 100107
wow_900 = true
wow_800 = true
wow_503 = true
maxSlots = 200
NUM_PER_ROW = 10
heightChange = nil
widthDelta = 315
heightDelta = 204
f = ImprovedStableFrameSlots {
 0 = <userdata>
 NineSlice = Frame {
 }
 layoutType = "InsetFrameTemplate"
 Bg = Texture {
 }
}
(*temporary) = nil
(*temporary) = nil
(*temporary) = 485
(*temporary) = <userdata>
(*temporary) = 438
(*temporary) = <userdata>
(*temporary) = -26.000002
(*temporary) = <userdata>
(*temporary) = 91.000008
(*temporary) = -26.000002
(*temporary) = <table> {
 BottomLeftCorner = <table> {
 }
 RightEdge = <table> {
 }
 TopEdge = <table> {
 }
 BottomEdge = <table> {
 }
 TopRightCorner = <table> {
 }
 TopLeftCorner = <table> {
 }
 LeftEdge = <table> {
 }
 BottomRightCorner = <table> {
 }
}
(*temporary) = nil
(*temporary) = nil
(*temporary) = <table> {
 1 = <table> {
 }
 2 = <table> {
 }
 3 = <table> {
 }
 4 = <table> {
 }
 5 = <table> {
 }
 6 = <table> {
 }
 7 = <table> {
 }
 8 = <table> {
 }
 9 = <table> {
 }
}
(*temporary) = 9
(*temporary) = nil
(*temporary) = "attempt to index global 'PetStableModel' (a nil value)"

**Comment by tflo** [comment]
Change `PetStableModel` to `PetStableModelScene` in _ImprovedStableFrame.lua_, lines 102 and 103:

```diff
--- Old	2023-09-06 15:23:47.000000000 
+++ New	2023-09-06 15:23:47.000000000 
@@ -96,14 +96,14 @@
 	PetStableFrame.Inset:SetPoint("TOPLEFT", f, "TOPRIGHT")
 	PetStableFrame:SetWidth(PetStableFrame:GetWidth() + widthDelta)
 	PetStableFrame:SetHeight(PetStableFrame:GetHeight() + heightDelta)
 
 	PetStableFrameModelBg:SetHeight(281 + heightDelta)
 
-	local p, r, rp, x, y = PetStableModel:GetPoint(1)
-	PetStableModel:SetPoint(p, r, rp, x, y - 32)
+	local p, r, rp, x, y = PetStableModelScene:GetPoint(1)
+	PetStableModelScene:SetPoint(p, r, rp, x, y - 32)
 
 	PetStableStabledPet1:ClearAllPoints()
 	PetStableStabledPet1:SetPoint("TOPLEFT", f, 8, -36)
 
 
 	local searchInput = CreateFrame("EditBox", "ISF_SearchInput", f, "SearchBoxTemplate")
```

**Comment by ZForceSeattle** [comment]
I have the same problem, but I thought I'd add a picture of what it looks like if I go to the stable master.  I thought it might be a simple update of the stable master data table, but I've tried it several times with no luck.  I have an addon that allows me to move my frames around, so I also tried to see if the reason was that it was not fitting onto my screen fully and was running into the null on the bottom part of the frame.  But it made no difference. After a reload, I got the same error. The small library file is not in the middle of the main frame of the addon. 

I've added a picture, but it is cutting off the bottom part due to a known issue with the Snipping Tool in Windows. 

![Improved Stable](https://github.com/ascott18/ImprovedStableFrame/assets/59213553/a732dee6-56dc-46fa-94db-2d2869f4c542)

**Comment by tflo** [comment]
Have you tried the fix I posted above?

For me, it works. I have only tested it briefly, but I can move pets around, exchange pets, and the frame representation is fixed. (I had the same error as the OP, and the same graphical glitch as in your screenshot.)

Here is a copy of the addon as I'm using it:

[ImprovedStableFrame.zip](https://github.com/ascott18/ImprovedStableFrame/files/12540705/ImprovedStableFrame.zip)

This version contains an additional change: It checks at load time if the player class is a Hunter. If not, it doesn't load any code. If you don't like that, just remove the check at the very beginning of _ImprovedStableFrame.lua_ (2 lines).

---

PS: I'm on Retail.

**Comment by ZForceSeattle** [comment]
### POSSIBLE SOLUTION

Thank you for your message. When did you get your update? I'm using the November 2022 release listed here and on Curseforge (app and page). I compared the TOC and the lua; they look the same, but....

I deleted the variables, directories, cache files, and everything related to the addon. I reinstalled the addon from the Curseforge APP, and even though it still shows a version date of 10.0 - November - It is now working. 

I am still determining where the problem comes from. I rarely need to open the stable frame because I always use the same pets. The reinstall also fixed an error I was getting on another addon - weird. 

I did get the "You are using out-of-date addons" message when I logged on today, and I'm wondering if it is becoming a thing with the version dates. My friend who does the primary programming with Titan said it had become a problem for some of his items.

**Comment by tflo** [comment]
> "You are using out-of-date addons"

This is completely irrelevant, it's just a number in the toc file that is checked by Blizz, not any relevant content. 

> When did you get your update?

Not sure what you mean. The latest version by the author of the addon (to my knowledge) is 10.0.0. I haven't seen an update since then.

> I am still determining where the problem comes from.

As said [here](https://github.com/ascott18/ImprovedStableFrame/issues/5#issuecomment-1708365438), it's a frame name that has changed in 10.1.7, from `PetStableModel` to `PetStableModelScene`. (Hence the "nil value" in the error.)

If you apply my fix (or download my patched version from [above](https://github.com/ascott18/ImprovedStableFrame/issues/5#issuecomment-1708804248)), then it should work. At least for me it does. No need to delete Saved Variables or cache files.

PS: In my patch, I've also updated the toc number for 10.1.7, so you won't get any "You are using out-of-date addons" warning either, in case this matters. I have left the version number of the addon unchanged (10.0.0).

**Comment by ascott18** [comment]
Fixed in [a8210c3](https://github.com/ascott18/ImprovedStableFrame/commit/a8210c30f34e99846e556de96c8d741fd0f2e390)
